app/cli.py

```diff
@@ -0,0 +1,235 @@
+
+
+def create_install_config_file(clusters, pull_secret_file):
+    ssh_key = get_local_ssh_key()
+    pull_secret = json.dumps(get_pull_secret_data(pull_secret_file=pull_secret_file))
+    for _cluster in clusters:
+        install_dir = _cluster["install-dir"]
+        _cluster["pull_secret"] = pull_secret
+        _cluster["ssh_key"] = ssh_key
+
+        cluster_install_config = get_install_config_j2_template(cluster_dict=_cluster)
+        Path(install_dir).mkdir(parents=True, exist_ok=True)
+
+        with open(os.path.join(install_dir, "install-config.yaml"), "w") as fd:
+            fd.write(yaml.dump(cluster_install_config))
+
+    return clusters
+
+
+def get_pull_secret_data(pull_secret_file):
+    with open(pull_secret_file) as fd:
+        return json.load(fd)
+
+
+def get_local_ssh_key():
+    with open(os.path.expanduser("~/.ssh/id_rsa.pub")) as fd:
+        return fd.read().strip()
+
+
+@click.command()
+@click.option("-i", "--install", is_flag=True, help="Install Openshift cluster/s")
+@click.option("-u", "--uninstall", is_flag=True, help="Uninstall Openshift cluster/s")
+@click.option(
+    "-p",
+    "--parallel",
+    help="Run clusters install/uninstall in parallel",
+    is_flag=True,
+    show_default=True,
+)
+@click.option(
+    "--clusters-install-data-directory",
+    help="""
+\b
+Path to cluster install data
+    For install this will be used to store the install data.
+    For uninstall this will be used to uninstall the cluster.
+""",
+    default=os.environ.get(
+        "CLUSTER_INSTALL_DATA_DIRECTORY",
+        "/openshift-cli-installer/clusters-install-data",
+    ),
+    type=click.Path(exists=True),
+    show_default=True,
+)
+@click.option(
+    "--pull-secret-file",
+    help="Path to pull secret json file, can be obtained from console.redhat.com",
+    required=True,
+    default=os.environ.get("PULL_SECRET"),
+    type=click.Path(exists=True),
+    show_default=True,
+)
+@click.option(
+    "-c",
+    "--cluster",
+    type=DictParamType(),
+    help="""
+\b
+Cluster/s to install.
+Format to pass is:
+    'name=cluster1;base_domain=aws.domain.com;platform=aws;region=us-east-2;version=4.14.0-ec.2'
+Required parameters:
+    name: Cluster name.
+    base_domain: Base domain for the cluster.
+    platform: Cloud platform to install the cluster on. (Currently only AWS supported).
+    region: Region to use for the cloud platform.
+    version: Openshift cluster version to install
+
+Check install-config-template.j2 for variables that can be overwritten by the user.
+For example:
+    fips=true
+    worker_flavor=m5.xlarge
+    worker_replicas=6
+    """,
+    required=True,
+    multiple=True,
+)
+def main(
+    install,
+    uninstall,
+    pull_secret_file,
+    parallel,
+    cluster,
+    clusters_install_data_directory,
+):
+    """
+    Install/Uninstall Openshift cluster/s
+    """
+    set_and_verify_aws_credentials()
+    if not (install or uninstall):
```

Comment:
Do we want to disable both being set? the code 'prioritize' install over uninstall if both are set to True (as flags)

Reply:
Or make co existence of install&uninstall processes in the code

Reply:
right, fixed.